novawallet/Modules/Staking/NominationPoolSearch/NominationPoolSearchPresenter.swift

```diff
@@ -0,0 +1,154 @@
+import Foundation
+import SoraFoundation
+import RobinHood
+
+final class NominationPoolSearchPresenter: AnyCancellableCleaning {
+    weak var view: NominationPoolSearchViewProtocol?
+    weak var delegate: StakingSelectPoolDelegate?
+
+    let wireframe: NominationPoolSearchWireframeProtocol
+    let interactor: NominationPoolSearchInteractorInputProtocol
+    let viewModelFactory: StakingSelectPoolViewModelFactoryProtocol
+    let chainAsset: ChainAsset
+    let searchOperationFactory: NominationPoolSearchOperationFactoryProtocol
+    let operationQueue: OperationQueue
+
+    private var poolStats: [NominationPools.PoolStats] = []
+    private var currentSearchOperation: CancellableCall?
+    private var searchOperationClosure: NominationPoolSearchOperationClosure?
+
+    init(
+        interactor: NominationPoolSearchInteractorInputProtocol,
+        wireframe: NominationPoolSearchWireframeProtocol,
+        viewModelFactory: StakingSelectPoolViewModelFactoryProtocol,
+        chainAsset: ChainAsset,
+        searchOperationFactory: NominationPoolSearchOperationFactoryProtocol,
+        delegate: StakingSelectPoolDelegate,
+        operationQueue: OperationQueue
+    ) {
+        self.interactor = interactor
+        self.wireframe = wireframe
+        self.viewModelFactory = viewModelFactory
+        self.chainAsset = chainAsset
+        self.searchOperationFactory = searchOperationFactory
+        self.operationQueue = operationQueue
+        self.delegate = delegate
+    }
+
+    private func provideVidewModel() {
+        let viewModel = viewModelFactory.createStakingSelectPoolViewModels(
+            from: poolStats,
+            selectedPoolId: nil,
+            chainAsset: chainAsset,
+            locale: selectedLocale
+        )
+        searchOperationClosure = searchOperationFactory.createOperationClosure(viewModels: viewModel)
+    }
+
+    private func showUnsupportedPoolStateAlert() {
+        let title = R.string.localizable.commonErrorGeneralTitle(preferredLanguages: selectedLocale.rLanguages)
+        let message = R.string.localizable.stakingSearchPoolSelectionErrorMessage(
+            preferredLanguages: selectedLocale.rLanguages)
+        let closeAction = R.string.localizable.commonClose(preferredLanguages: selectedLocale.rLanguages)
+        wireframe.present(message: message, title: title, closeAction: closeAction, from: view)
+    }
+}
+
+extension NominationPoolSearchPresenter: NominationPoolSearchPresenterProtocol {
+    func setup() {
+        interactor.setup()
+        view?.didReceivePools(state: .loaded(viewModel: []))
+    }
+
+    func search(for text: String) {
+        guard let searchOperationClosure = searchOperationClosure else {
+            return
+        }
+        if text.isEmpty {
```

Comment:
Looks like logic for the interactor

Reply:
why? We search in viewModels

Reply:
It is not trivial and performs asyncroniously